packages/semcom-demo-app/src/app/home/home.effects.ts

```diff
@@ -27,7 +27,7 @@ export class HomeEffects {
   queryMetadataFromShapes$ = createEffect(() => this.actions$.pipe(
     ofType(shapesDetected),
     mergeMap(({shapeIds}) => forkJoin(shapeIds.concat([ 'http://digita.ai/voc/input#input' ]).map((shapeId) => this.semComService.queryComponents(shapeId)))),
-    map((resultsPerShape) => resultsPerShape.filter((results) => results.length > 0)),
+    map((resultsPerShape) => resultsPerShape.filter((results) => results.length > 0 && results[0])),
```

Comment:
Instead of checking here, we should rewrite the query function to never return undefined.